### DISCUSSION TITLE
Add support for "dot"

### DISCUSSION THREAD

**Discussion by ZekyTheWolf** [nhedger/vscode-inertia#25] [Ideas]
Add support for . instead of just / in inertia function ?

**Comment by nhedger** [comment]
Hey, thanks for reaching out.

I'm assuming that you mean dot notation for component names. Does that even work with Inertia ?

Found [this issue](https://github.com/inertiajs/inertia-laravel/issues/48) suggesting that they did not implement it.

**Reply by ZekyTheWolf** [reply]
Yes, it works.

Here is example for app.ts that i use ( [original](https://laravel-vite.netlify.app/) that i was using before laravel migrated to vite)

```ts
// @ts-ignore
import { createInertiaApp } from "@inertiajs/svelte";

import "../css/app.css";

createInertiaApp({
    progress: {
        color: "#29d",
    },
    // @ts-ignore
    resolve: (name) =>
        importPageComponent(name, import.meta.glob("../views/**/*.svelte")),

    // @ts-ignore
    setup({ el, App, props }) {
        new App({ target: el, props });
    },
});

function importPageComponent(name: string, pages: Record<string, any>) {
    // eslint-disable-next-line no-restricted-syntax
    for (const path in pages) {
        if (path.endsWith(`${name.replaceAll(".", "/")}.svelte`)) {
            return typeof pages[path] === "function"
                ? pages[path]()
                : pages[path];
        }
    }

    throw new Error(`Page not found: ${name}`);
}
```

**Reply by nhedger** [reply]
Ah, I see you're replacing it by hand, but Inertia still uses `/` separators in the end.

I'll try to think of something, but I'm reasonably confident it can be done.

One thing to note though, on the backend using `.` will not work AFAIK.

**Reply by ZekyTheWolf** [reply]
Im fairly okey with this, just having option for opening files from `inertia()` function when using `.` would be nice and very helpfull.

**Comment by nhedger** [comment]
I have just released version `0.4`, which should handle your use case.

See the `inertia.pathSeparators` setting.

The relevant documentation section was added to the README as well.

**Reply by ZekyTheWolf** [reply]
Working, big thanks.

**Comment by ZekyTheWolf** [comment]
Only when creating file via this ext. it create file.name.svelte instead of ../file/name.svelte

**Reply by nhedger** [reply]
Good catch; thanks for reporting.

Can you update to `0.4.2` a confirm that it now works ?

**Reply by ZekyTheWolf** [reply]
Working as should, nice job 👍

**Reply by nhedger** [reply]
Good to hear !
